src/main/groovy/edu/oregonstate/mist/spendingauthority/db/SpendingAuthorityDAOWrapper.groovy

```diff
@@ -0,0 +1,77 @@
+package edu.oregonstate.mist.spendingauthority.db
+
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import edu.oregonstate.mist.spendingauthority.Attributes
+import edu.oregonstate.mist.spendingauthority.LimitsForIndexes
+import edu.oregonstate.mist.spendingauthority.SpendingLimit
+
+class SpendingAuthorityDAOWrapper {
+    private final SpendingAuthorityDAO spendingAuthorityDAO
+
+    private final BigDecimal requestorLimit = 4999.99
+
+    SpendingAuthorityDAOWrapper(SpendingAuthorityDAO spendingAuthorityDAO) {
+        this.spendingAuthorityDAO = spendingAuthorityDAO
+    }
+
+    public ResultObject getSpendingLimits(String onid) {
+        new ResultObject(data: getResourceObject(onid))
+    }
+
+    private ResourceObject getResourceObject(String onid) {
+        // Query expects onid in upper case
+        String upperCaseOnid = onid.toUpperCase()
+
+        // Queues for approvers. Approvers are tied to queues, which are tied to indexes.
+        List<SpendingLimit> spendingLimits = spendingAuthorityDAO.getQueues(upperCaseOnid)
+
+        // Indexes for requestors. Requestors have one spending limit.
+        List<String> requestorIndexes = spendingAuthorityDAO.getRequestorIndexes(upperCaseOnid)
+
+        if (!spendingLimits && !requestorIndexes) {
+            return null
+        }
+
+        def limitsMap = [:]
+
+        if (spendingLimits) {
+            spendingLimits.each {
```

Comment:
I think we can use the Groovy null safe operator here: `spendingLimits?.each {...}`